### PR TITLE
Close #12242: Convert TrackDesignPlace static globals to use Coords

### LINKED ISSUE #12242: Convert TrackDesignPlace static globals to use Coords
There are a handful of global static variables in `TrackDesignPlace.cpp` that store coordinates, but are not coupled. The task as hand is to merge them into just a few variables:

- [ ] `_window_track_place_last_x` and `_window_track_place_last_y` should become a `CoordsXY _windowTrackPlaceLast`. Use `setNull()` instead of ` = -1` where applicable.
- [ ] `_window_track_place_last_valid_x`, `_window_track_place_last_valid_y`, `_window_track_place_last_valid_z` should become `CoordsXYZ _windowTrackPlaceLastValid`

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 1]
```diff
@@ -171,7 +169,7 @@ rct_window* window_track_place_open(const track_design_file_ref* tdFileRef)
     window_push_others_right(w);
     show_gridlines();
     _window_track_place_last_cost = MONEY32_UNDEFINED;
-    _window_track_place_last_x = -1;
+    _windowTrackPlaceLast.x = -1;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp:172]:
```suggestion
    _windowTrackPlaceLast.setNull();
```

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 2]
```diff
@@ -212,14 +210,14 @@ static void window_track_place_mouseup(rct_window* w, rct_widgetindex widgetInde
             window_track_place_clear_provisional();
             _currentTrackPieceDirection = (_currentTrackPieceDirection + 1) & 3;
             w->Invalidate();
-            _window_track_place_last_x = -1;
+            _windowTrackPlaceLast.x = -1;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp:213]:
```suggestion
            _windowTrackPlaceLast.setNull();
```

File: src/openrct2-ui/windows/TrackDesignPlace.cpp [hunk 3]
```diff
@@ -212,14 +210,14 @@ static void window_track_place_mouseup(rct_window* w, rct_widgetindex widgetInde
...
             window_track_place_draw_mini_preview(_trackDesign.get());
             break;
         case WIDX_MIRROR:
             track_design_mirror(_trackDesign.get());
             _currentTrackPieceDirection = (0 - _currentTrackPieceDirection) & 3;
             w->Invalidate();
-            _window_track_place_last_x = -1;
+            _windowTrackPlaceLast.x = -1;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/TrackDesignPlace.cpp:220]:
```suggestion
            _windowTrackPlaceLast.setNull();
```

**Comment by duncanspumpkin** [comment]
After using set null make sure you test this a bit as the functions called I'm pretty sure do not have the corresponding isNull

**Comment by tupaschoal** [comment]
This is all good, but can you please rename the first commit to "Part of #12242: ..."? 

Sorry for not pointing it out earlier.

**Comment by Gymnasiast** [comment]
@tupaschoal I just squashed them. It wasn't much code anyway.

**Pull request merged**